sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachinesImpl.java

```diff
@@ -162,9 +163,9 @@ public Mono<String> captureAsync(
             .map(
                 captureResultInner -> {
                     try {
-                        ObjectMapper mapper = new ObjectMapper();
-                        return mapper.writeValueAsString(captureResultInner);
-                    } catch (JsonProcessingException ex) {
+                        return SerializerFactory.createDefaultManagementSerializerAdapter()
+                            .serialize(captureResultInner, SerializerEncoding.JSON);
+                    } catch (IOException ex) {
                         throw logger.logExceptionAsError(Exceptions.propagate(ex));
                     }
```

Comment:
Why this is different from above?
https://github.com/Azure/azure-sdk-for-java/pull/41407/commits/c1f87f7e96d7ffb6c9f8e549e65c91d9757e7583#r1701550499

Reply:
I can't see what codesnippet the link is pointing to.. 

Could you paste the code?

Reply:
this https://github.com/Azure/azure-sdk-for-java/pull/41407#discussion_r1701550499

Reply:
Good catch.. Yes, they are the same. 

Seems previous implementation is wrong. I'll fix here.

Reply:
fixed